test/miner_state_channel_SUITE.erl

```diff
@@ -0,0 +1,546 @@
+    InitialDCTxns = [blockchain_txn_dc_coinbase_v1:new(Addr, Balance) || Addr <- Addresses],
+    AddGwTxns = [blockchain_txn_gen_gateway_v1:new(Addr, Addr, h3:from_geo({37.780586, -122.469470}, 13), 0)
+                 || Addr <- Addresses],
+
+    NumConsensusMembers = ?config(num_consensus_members, Config),
+    BlockTime = ?config(block_time, Config),
+    BatchSize = ?config(batch_size, Config),
+    Curve = ?config(dkg_curve, Config),
+
+    SCVars = #{max_open_sc => 2,
+               min_expire_within => 10,
+               max_xor_filter_size => 1024*100,
+               max_xor_filter_num => 5,
+               max_subnet_size => 65536,
+               min_subnet_size => 8,
+               max_subnet_num => 20,
+               sc_grace_blocks => 5},
+
+    DefaultVars = #{?block_time => BlockTime,
+                    %% rule out rewards
+                    ?election_interval => infinity,
+                    ?num_consensus_members => NumConsensusMembers,
+                    ?batch_size => BatchSize,
+                    ?dkg_curve => Curve},
+
+    Keys = libp2p_crypto:generate_keys(ecc_compact),
+
+    InitialVars = miner_ct_utils:make_vars(Keys, maps:merge(DefaultVars, SCVars)),
+
+    DKGResults = miner_ct_utils:inital_dkg(Miners,
+                                           InitialVars ++ InitialPaymentTransactions ++ AddGwTxns ++ InitialDCTxns,
+                                           Addresses, NumConsensusMembers, Curve),
+    true = lists:all(fun(Res) -> Res == ok end, DKGResults),
+
+    %% Get both consensus and non consensus miners
+    {ConsensusMiners, NonConsensusMiners} = miner_ct_utils:miners_by_consensus_state(Miners),
+    %% integrate genesis block
+    _GenesisLoadResults = miner_ct_utils:integrate_genesis_block(hd(ConsensusMiners), NonConsensusMiners),
+
+    %% confirm we have a height of 1
+    ok = miner_ct_utils:wait_for_gte(height_exactly, Miners, 1),
+
+    [{consensus_miners, ConsensusMiners},
+     {non_consensus_miners, NonConsensusMiners},
+     {default_routers, application:get_env(miner, default_routers, [])}
+     | Config].
+
+end_per_testcase(_TestCase, Config) ->
+    miner_ct_utils:end_per_testcase(_TestCase, Config).
+
+no_packets_expiry_test(Config) ->
+    Miners = ?config(miners, Config),
+
+    [RouterNode | _] = Miners,
+
+    %% setup
+    %% oui txn
+    {ok, RouterPubkey, RouterSigFun, _ECDHFun} = ct_rpc:call(RouterNode, blockchain_swarm, keys, []),
+    RouterPubkeyBin = libp2p_crypto:pubkey_to_bin(RouterPubkey),
+    RouterSwarm = ct_rpc:call(RouterNode, blockchain_swarm, swarm, []),
+    ct:pal("RouterSwarm: ~p", [RouterSwarm]),
+    RouterP2PAddress = ct_rpc:call(RouterNode, libp2p_swarm, p2p_address, [RouterSwarm]),
+    ct:pal("RouterP2PAddress: ~p", [RouterP2PAddress]),
+
+    ct:pal("Height before oui: ~p", [miner_ct_utils:height(RouterNode)]),
+
+    EUIs = [{16#deadbeef, 16#deadc0de}],
+    {Filter, _} = xor16:to_bin(xor16:new([ <<DevEUI:64/integer-unsigned-little,
```

Comment:
Could you explain what the filter does here?

Reply:
Put a comment there